circle-mlir/circle-mlir/lib/dialect/src/ConstFold.h

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// from tensorflow/compiler/mlir/lite/ir/tfl_ops.cc
+
+#ifndef __CIRCLE_MLIR_DIALECT_CONST_FOLD_H__
+#define __CIRCLE_MLIR_DIALECT_CONST_FOLD_H__
+
+#include "circle-mlir/dialect/CircleDialect.h"
+
+namespace mlir
+{
+namespace Circle
+{
+
+namespace
+{
```

Comment:
Using an anonymous namespace in a header file is generally not recommended. The purpose of an anonymous namespace is to provide internal linkage for symbols within a single translation unit. When placed in a header file, each translation unit that includes the header will have its own copy of the symbols defined in the anonymous namespace. This can lead to multiple definitions and might cause unexpected behavior or increased binary size. It's better to use "static" for functions or variables in a header if you intend internal linkage, or better yet, move those definitions to a source file.

Reply:
this header is only included from `CircleDialect.cpp` to reduce `CircleDialect.cpp` lines.
It's a bit weird for to use `static` after using anonymouse `namespace` with C++.
Do we really need to change?

Reply:
> > this header is only included from CircleDialect.cpp to reduce CircleDialect.cpp lines.

Although no error occurs if the header is used in only one source file, from a maintenance perspective it doesn't seem like a good practice. However, it would not be a problem for a while.

> > It's a bit weird for to use static after using anonymouse namespace with C++.
Do we really need to change?

I mentioned 'static' because of multiple definitions. Using an anonymous namespace in a header file does not prevent these problems. How about separating those functions into a source file as `ConstFold.cpp`? I think it's the best way.

Reply:
I've changed to `ConstFold.inc.cpp` :)

Reply:
@seanshpark 
After we talked about i, I understand this file will keep on going to be used in only one source file. Thanks for sharing your information.